arch-x86: raise #DE on division overflow

nmosier opened the pull request:
Fix #2730. Check for overflow when performing division. If the quotient does not fit in 8/16/32/64 bits for a 8-/16-/32-/64-bit div2 micro-op, then raise a divide error (#DE).

Change-Id: Ic06714ef9546c3542316ec3a3a74042fb613da64

powerjg reviewed (dismissed):
Thanks for fixing this!

nmosier committed:
arch-x86: raise #DE on division overflow

Fix #2730. Check for overflow when performing division.
If the quotient does not fit in 8/16/32/64 bits for
a 8-/16-/32-/64-bit div2 micro-op, then raise a divide
error (#DE).

Change-Id: Ida57c919f1eddff2ebc91a0c797d18e84c477c73
